sql/src/SqlEval.sk

```diff
@@ -475,8 +475,63 @@ class Evaluator{options: Options, user: ?UserFile} {
         dirDescr = computeTableDescr(name, schema.columns);
         this.createTable(context, pos, name, dirDescr, ifNotExists);
       })
-    | P.CreateTableAs _ ->
-      invariant_violation("CREATE TABLE AS SELECT not implemented")
+    | P.CreateTableAs{pos, name, select, ifNotExists} ->
+      core: P.SelectCoreQuery = select.core match {
+      | P.SelectCoreValues _ ->
+        error(
+          pos,
+          "CREATE TABLE AS requires a SELECT query, not a VALUES literal",
+        )
+      | core @ P.SelectCoreQuery _ -> core
+      };
+
+      compiler = SKDB.Compiler::create(false, this.options, select.pos, params);
+      cselect = compiler.compileSelect(
+        context,
+        select,
+        genSelectId(select),
+        true,
+      );
+
+      (_, cols) = SKDB.Compiler::getCols(core.params, cselect.from);
+
+      schema = cselect.params.mapWithIndex((idx, param) -> {
+        P.ColumnDefinition{
+          name => cols.maybeGet(idx) match {
+          | None() ->
+            error(
+              pos,
+              "Anonymous columns not allowed in CREATE TABLE AS SELECT",
```

Comment:
SQLite also accepts anonymous expressions here, and simply sets the column name to the string representation of the expression:
```
CREATE TABLE foo AS SELECT 2+2;
# Is equivalent to:
# CREATE TABLE foo ("2+2" INT);
# INSERT INTO foo VALUES(2+2);
```

Reply:
Right, I thought that being more restrictive than sqlite made sense for the schema migration application -- one can imagine mistakenly creating a table with anonymous columns and then being stuck with it.

But I can easily remove those restrictions for general uses and just enforce non-anonymous columns for schema migrations in particular.

Reply:
I think it's fine as is, especially since it's failing with a clear error message.